### PR TITLE
[ACQ-554] Filter section fixes

### PR DESCRIPTION
**Pull request by levadadenys**
### [ACQ-554] Filter section fixes
Fixes issues brought up in https://github.com/code-dot-org/code-dot-org/pull/52038
* fix dropdownButton focus state
* fix verticalAlign of filter by section
* fix button collapsing on small screen
* fix doubleclick select option text
* set max-height to dropdown options container
* adjustements for max-height to work correctly
* updated the alignment of multiline labels 

Before:

https://github.com/code-dot-org/code-dot-org/assets/22244040/b144d4c1-2f4a-46cb-aa5e-0cbaec7467eb

After:

https://github.com/code-dot-org/code-dot-org/assets/22244040/15f13c9a-4441-4890-b516-9b34708e6d98

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/checkbox-dropdown.module.scss [hunk 1]
```diff
@@ -63,6 +65,9 @@ form.dropDownMenuContainer {
 
       span {
         margin: 0;
+        -webkit-user-select: none;
```
Comment (megcrenshaw) [inline apps/src/templates/checkbox-dropdown.module.scss:79]: I'm seeing here https://developer.mozilla.org/en-US/docs/Web/CSS/user-select that the webkit-user-select is necessary but not the others. Is there a reason to specific moz and ms too?

File: apps/src/templates/checkbox-dropdown.module.scss [hunk 2]
```diff
@@ -7,13 +7,15 @@ button.dropdownButton {
...
+  display: flex;
+  align-items: baseline;
 
   &:hover {
     background-color: $neutral_dark10;
   }
 
   &:focus {
-    outline: 2px solid $brand_primary_light;
+    outline: 2px solid $light_primary_500;
```
Comment (megcrenshaw) [inline apps/src/templates/checkbox-dropdown.module.scss:18]: 🙌 :accessibility:

**Comment by megcrenshaw** [comment]
This is looking great! Two small nits:
1) The text (for example, "Interdisciplinary") is not vertically aligned with the checkbox
image
2) When the dropdown goes below the page, would it be easy to allow that dropdown to scroll down without the whole page scrolling?
image

**Comment by megcrenshaw** [comment]
I'm afraid I can still highlight the text (using Chrome):
image
Is that supposed to be fixed here?

File: apps/src/templates/checkbox-dropdown.module.scss (hunk 1 above)
Comment (levadadenys) [reply apps/src/templates/checkbox-dropdown.module.scss:79]: `-webkit-user-select` and `user-select` are supported in most of the browsers, however:

`-moz-user-select` - is needed for support of firefox 69- - 1.5+ versions
`-ms-user-select` - is needed for support of IE 10 & 11
Are we supporting those two?

**Comment by levadadenys** [comment]
@megcrenshaw as for text selection and vertical alignment (two comments above with screenshots) - are you sure you've tested it on the right branch? Or maybe there was some glitch with grunt/webpack on your machine? The styles of checkbox looks like you've been testing https://github.com/code-dot-org/code-dot-org/pull/52038 (if yes - sorry for confusion with branch names) 
Here's the video I've just made on my machine on `denys/acq/acq-554-2` branch. You can see checkbox icon has different styles alongside with

https://github.com/code-dot-org/code-dot-org/assets/22244040/d0c08194-9795-4573-9e31-8ec7ebe61fba

**Comment by levadadenys** [comment]
@megcrenshaw re `2) When the dropdown goes below the page, would it be easy to allow that dropdown to scroll down without the whole page scrolling`
That's a good question. We don't have it in designs currently, so it's something I need to discuss with design team first. We can set a max height for the dropdown and show scroll if it exceeds that max height, also there might be a way to make "on flight" calculations of that, but still I think it's better to discuss with design (rebranding) team first.

So I suggest we keep it as is right now unless it's urgent (adding `overflow-y: auto; max-height: "anyMagicNumber"px/vh;` won't take much time), and once design team will give it's response - implement that in DSCO_ version of dropdown

**Comment by megcrenshaw** [comment]
> So I suggest we keep it as is right now unless it's urgent (adding `overflow-y: auto; max-height: "anyMagicNumber"px/vh;` won't take much time), and once design team will give it's response - implement that in DSCO_ version of dropdown

That sounds great –– it's not urgent!

> are you sure you've tested it on the right branch? Or maybe there was some glitch with grunt/webpack on your machine? The styles of checkbox looks like you've been testing https://github.com/code-dot-org/code-dot-org/pull/52038 (if yes - sorry for confusion with branch names)

Yikes, so sorry, Denys! I thought I checked, but apparently not –– thanks for the nudge to check. I am seeing inconsistent behavior on "click" and on keyboard "space": the "click" shows another state with a dark background but the "space" does not. Any chance we could get rid of that state that has the dark background?

https://github.com/code-dot-org/code-dot-org/assets/9142121/ab6c6d99-7c6f-44b5-a037-1de13b7bf957

File: apps/src/templates/checkbox-dropdown.module.scss (hunk 1 above)
Comment (megcrenshaw) [reply apps/src/templates/checkbox-dropdown.module.scss:79]: We don't support IE anymore, and we only support Firefox 91.x! https://code.org/educate/it

Comment (levadadenys) [reply apps/src/templates/checkbox-dropdown.module.scss:79]: Cool, will delete those two then. Thank you!

**Comment by levadadenys** [comment]
> > So I suggest we keep it as is right now unless it's urgent (adding `overflow-y: auto; max-height: "anyMagicNumber"px/vh;` won't take much time), and once design team will give it's response - implement that in DSCO_ version of dropdown
> 
> That sounds great –– it's not urgent!
> 
> > are you sure you've tested it on the right branch? Or maybe there was some glitch with grunt/webpack on your machine? The styles of checkbox looks like you've been testing #52038 (if yes - sorry for confusion with branch names)
> 
> Yikes, so sorry, Denys! I thought I checked, but apparently not –– thanks for the nudge to check. I am seeing inconsistent behavior on "click" and on keyboard "space": the "click" shows another state with a dark background but the "space" does not. Any chance we could get rid of that state that has the dark background?
> 
>  Screen.Recording.2023-06-15.at.9.23.31.AM.mov

It's pressed state and it's part of the designs. The actual problem here is that it currently has no pressed state when space is pressed.

**Comment by levadadenys** [comment]
Fixed pressed state, clenedup user-select.

**Comment by levadadenys** [comment]
Moved text selection update to checkbox component in separate RP: https://github.com/code-dot-org/code-dot-org/pull/52337

**Comment by levadadenys** [comment]
@megcrenshaw @breville talked with @markabarrett about dropdown components. Since we're aiming desktop first - we will use custom dropDowns instead of native ones. I'll take a look if it's possible to make some workaround for mobile devices though once I'll be creating to DSCO_ dropdown components, but it'll be lower priority.

As of max-height - we agreed dropdowns should have it, will add it in the next commit

**Comment by breville** [comment]
> Since we're aiming desktop first - we will use custom dropDowns instead of native ones. I'll take a look if it's possible to make some workaround for mobile devices though once I'll be creating to DSCO_ dropdown components, but it'll be lower priority.

I'm slightly concerned that we are only differentiating between desktop and mobile.  There is a wide spectrum of devices in between the two, such as touch-screen laptops, tablets, etc.  I think we lose a lot of accessibility across the full range of target devices once we do a custom implementation of something as core as a dropdown. 

Also, have we evaluated the impact on things like screenreaders with a custom implementation instead of using a browser-native control? 

I also don't feel like we need to do custom styling of native browser controls like dropdowns.  It might not be as pretty, but it lets the browser offer a consistent experience to users that is appropriate to their operating system.

**Comment by levadadenys** [comment]
> @megcrenshaw @breville talked with @markabarrett about dropdown components. Since we're aiming desktop first - we will use custom dropDowns instead of native ones. I'll take a look if it's possible to make some workaround for mobile devices though once I'll be creating to DSCO_ dropdown components, but it'll be lower priority.
> 
> As of max-height - we agreed dropdowns should have it, will add it in the next commit

Done.
Here's video:

https://github.com/code-dot-org/code-dot-org/assets/22244040/3c85d823-f1dd-4eb6-adf9-200061e38723

**Comment by breville** [comment]
> Done.

Thanks for the video.  I think it feels like "Select all" and "Clear all" should always be visible, pinned to the bottom of the available dropdown, and the list of content should scroll inside of that?

P.S. apologies for the accidental close below; pressed the wrong button!

**Comment by levadadenys** [comment]
> > Since we're aiming desktop first - we will use custom dropDowns instead of native ones. I'll take a look if it's possible to make some workaround for mobile devices though once I'll be creating to DSCO_ dropdown components, but it'll be lower priority.
> 
> I'm slightly concerned that we are only differentiating between desktop and mobile. There is a wide spectrum of devices in between the two, such as touch-screen laptops, tablets, etc. I think we lose a lot of accessibility across the full range of target devices once we do a custom implementation of something as core as a dropdown.
> 
> Also, have we evaluated the impact on things like screenreaders with a custom implementation instead of using a browser-native control?
> 
> I also don't feel like we need to do custom styling of native browser controls like dropdowns. It might not be as pretty, but it lets the browser offer a consistent experience to users that is appropriate to their operating system.

Agree, there's space to discuss still. However, it's something to be discussed with design team and we're not implementing dsco version of dropdown yet. This changes are urgent tohugh, so I suggest  merging this pr and raise that question with @markabarrett while developing DSCO_ version of dropdown. (Current checkbox dropdown is the one @code-dot-org/teacher-tools (if I remember team correctly)  implemented for work on new screens their developing, dsco_ will replace it once it'll be done)

**Comment by levadadenys** [comment]
> > Done.
> 
> Thanks for the video. I think it feels like "Select all" and "Clear all" should always be visible, pinned to the bottom of the available dropdown, and the list of content should scroll inside of that?
> 
> P.S. apologies for the accidental close below; pressed the wrong button!

oh, you got me shocked by closing 😆 
Re select/clear all buttons - good point, need to discuss it with design team

**Review by megcrenshaw — APPROVED** [review]
+1 to Brendan's questions and concerns about screen size and the "Select All" and "Clear All" buttons being pinned. I'm okay with other features being in a separate PR, after discussion with the design team. I know, Denys, you're navigating a lot going back and forth between us and the designers.

Overall, this is working and feeling great when I test it locally! Thank you for the demo too. Excited about this!

**Comment by breville** [comment]
> Since we're aiming desktop first - we will use custom dropDowns instead of native ones.

I should correct myself: there doesn't seem not be a great browser-native way of having drop downs that contain multiple-selection options.

**Comment by levadadenys** [comment]
> > > Done.
> > 
> > 
> > Thanks for the video. I think it feels like "Select all" and "Clear all" should always be visible, pinned to the bottom of the available dropdown, and the list of content should scroll inside of that?
> > P.S. apologies for the accidental close below; pressed the wrong button!
> 
> oh, you got me shocked by closing 😆 Re select/clear all buttons - good point, need to discuss it with design team

I've tried it locally today and it turned out so much better that I'll go ahead and push it in this pr.

https://github.com/code-dot-org/code-dot-org/assets/22244040/e99dbaaf-39ae-4330-aacc-e656b2791cbe

**Comment by levadadenys** [comment]
Talked with Serhii from design team, also updated the alignment of multiline labels (sse games and animations)
Знімок екрана 2023-06-21 о 18 08 24

**Pull request merged**